UnityProject/Assets/Scripts/Weapons/WeaponNetworkActions.cs

```diff
@@ -158,12 +160,79 @@ public void CmdUnloadWeapon(GameObject gunObject)
 		victimHealth.ApplyDamage(gameObject, (int) weaponAttr.hitDamage, DamageType.Brute, damageZone);
 		if (weaponAttr.hitDamage > 0)
 		{
-			PostToChatMessage.SendItemAttackMessage(weapon, gameObject, victim, (int) weaponAttr.hitDamage, damageZone);
+			PostToChatMessage.SendAttackMessage(gameObject, victim, (int) weaponAttr.hitDamage, damageZone, weapon);
 		}
 
 		SoundManager.PlayNetworkedAtPos( weaponAttr.hitSound, transform.position );
 		StartCoroutine(AttackCoolDown());
+	}
+
+	/// <summary>
+	/// Performs a punch attempt from one player to a target.
+	/// </summary>
+	/// <param name="punchDirection">The direction of the punch towards the victim.</param>
+	/// <param name="damageZone">The part of the body that is being punched.</param>
+	[Command]
+	public void CmdRequestPunchAttack(GameObject victim, Vector2 punchDirection, BodyPartType damageZone,
+		string attackerName)
+	{
+		var victimHealth = victim.GetComponent<LivingHealthBehaviour>();
+		var victimRegisterTile = victim.GetComponent<RegisterTile>();
+		var rng = new System.Random();
+
+		if (!playerScript.IsInReach(victim, true))
+		{
+			return;
+		}
 
+		// If the punch is not self inflicted, do the simple lerp attack animation.
+		if (victim != gameObject)
+		{
+			RpcMeleeAttackLerp(punchDirection, null);
+			playerMove.allowInput = false;
+		}
+
+		// This is based off the alien/humanoid/attack_hand punch code of TGStation's codebase.
+		// Punches have 90% chance to hit, otherwise it is a miss.
+		if (90 >= rng.Next(1, 100))
+		{
+			// The punch hit.
+			victimHealth.ApplyDamage(gameObject, (int) fistDamage, DamageType.Brute, damageZone);
+			if (fistDamage > 0)
+			{
+				PostToChatMessage.SendAttackMessage(gameObject, victim, (int) fistDamage, damageZone);
+			}
+
+			// Make a random punch hit sound.
+SoundManager.PlayNetworkedAtPos("Punch"+rng.Next(1, 4), victimRegisterTile.WorldPosition);
+			{
+				case 1:
+					SoundManager.PlayNetworkedAtPos("Punch1", victimRegisterTile.WorldPosition);
+					break;
+				case 2:
+					SoundManager.PlayNetworkedAtPos("Punch2", victimRegisterTile.WorldPosition);
+					break;
+				case 3:
+					SoundManager.PlayNetworkedAtPos("Punch3", victimRegisterTile.WorldPosition);
+					break;
+				case 4:
+					SoundManager.PlayNetworkedAtPos("Punch4", victimRegisterTile.WorldPosition);
+					break;
+			}
+
+			StartCoroutine(AttackCoolDown());
+		}
+		else
+		{
+			// The punch missed.
+			string victimName = victim.Player()?.Name;
+			SoundManager.PlayNetworkedAtPos("PunchMiss", transform.position);
+			ChatRelay.Instance.AddToChatLogServer(new ChatEvent
+			{
+				channels = ChatChannel.Local,
+				message = $"{attackerName} has attempted to punch {victimName}!"
```

Comment:
apparently attackerName is empty
(and is it needed at all?)
![img](https://user-images.githubusercontent.com/10403536/60588173-c30b0000-9d85-11e9-96f0-5cd13d24a06f.png)

Reply:
![0mqzssR](https://user-images.githubusercontent.com/28093344/60644714-08652580-9e8b-11e9-98b8-ad739a7c149a.png)
Fixed in next commit